File: php/tests/Unit/Application/UseCase/CreateGreetingUseCaseTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Application\UseCase;

use Alxarafe\App\Application\UseCase\CreateGreetingUseCase;
use Alxarafe\App\Infrastructure\Persistence\InMemoryGreetingRepository;
use PHPUnit\Framework\TestCase;

final class CreateGreetingUseCaseTest extends TestCase
{
    public function testExecuteCreatesGreeting(): void
    {
        $repository = new InMemoryGreetingRepository();
        $useCase = new CreateGreetingUseCase($repository);

        $greeting = $useCase->execute('World');

        self::assertSame('Hello, World!', $greeting->getMessage());
        self::assertNotEmpty($greeting->getId());
        self::assertInstanceOf(\DateTimeImmutable::class, $greeting->getCreatedAt());
    }

    public function testExecutePersistsGreeting(): void
    {
        $repository = new InMemoryGreetingRepository();
        $useCase = new CreateGreetingUseCase($repository);

        $greeting = $useCase->execute('Test');

        $found = $repository->findById($greeting->getId());
        self::assertNotNull($found);
        self::assertSame('Hello, Test!', $found->getMessage());
    }
}

File: java/src/test/java/com/alxarafe/application/usecase/CreateGreetingUseCaseTest.java
package com.alxarafe.application.usecase;

import com.alxarafe.app.application.usecase.CreateGreetingUseCase;
import com.alxarafe.app.domain.model.Greeting;
import com.alxarafe.app.infrastructure.persistence.InMemoryGreetingRepository;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CreateGreetingUseCaseTest {

    @Test
    void executeCreatesGreeting() {
        var repository = new InMemoryGreetingRepository();
        var useCase = new CreateGreetingUseCase(repository);

        var greeting = useCase.execute("World");

        assertEquals("Hello, World!", greeting.getMessage());
        assertNotNull(greeting.getId());
        assertNotNull(greeting.getCreatedAt());
    }

    @Test
    void executePersistsGreeting() {
        var repository = new InMemoryGreetingRepository();
        var useCase = new CreateGreetingUseCase(repository);

        var greeting = useCase.execute("Test");

        var found = repository.findById(greeting.getId());
        assertTrue(found.isPresent());
        assertEquals("Hello, Test!", found.get().getMessage());
    }
}
